covenant-signer/README.md

```diff
@@ -0,0 +1,301 @@
+
+```shell
+Derived private key: fe1c56c494c730f13739c0655bf06e615409870200047fc65cdf781837cf7f06
+Derived public key: 023a79b546c79d7f7c5ff20620d914b5cf7250631d12f6e26427ed9d3f98c5ccb1
+```
+
+As seen, the **Derived Public Key**:
+
+```
+023a79b546c79d7f7c5ff20620d914b5cf7250631d12f6e26427ed9d3f98c5ccb1
+```
+
+Matches the public key obtained earlier using the `getaddressinfo` command.
+
+#### 4. Importing the Derived Private Key into the Cosmos Keyring
+
+The derived private key can now be imported into the Cosmos keyring. Use the 
+following command:
+
+```shell
+babylond keys import-hex cov fe1c56c494c730f13739c0655bf06e615409870200047fc65cdf781837cf7f06
+```
+
+To confirm that the import was successful, run:
+
+```shell
+babylond keys show cov
+```
+
+The output will display the details of the imported key:
+
+```shell
+    address: bbn1azasawj3ard0ffwj04zpxlw2pt9cp7kwjcdqmc
+    name: cov
+    pubkey: '{"@type":"/cosmos.crypto.secp256k1.PubKey","key":"Ajp5tUbHnX98X/IGINkUtc9yUGMdEvbiZCftnT+Yxcyx"}'
+    type: local
+
+```
+
+Here, the `key` field contains the base64-encoded public key. After decoding, 
+this key:
+
+```shell
+023a79b546c79d7f7c5ff20620d914b5cf7250631d12f6e26427ed9d3f98c5ccb1
+```
+
+Matches the public key derived earlier and seen in the outputs of `getaddressinfo` 
+and `derive-child-key`.
+
+Congratulations! You have successfully imported your keys from the prior setup 
+and verified your setup for the covenant emulator.
+
+## 5. Create the configuration file
+
+Use the example configuration [file](./example/config.toml) to create your own 
+configuration file. Then, replace the placeholder values with your own configuration.
+
+```toml
+[keystore]
+keystore-type = "cosmos"
+
+[keystore.cosmos]
+# pointing to the directory where the key is stored, unless specified otherwise
+key-directory = "/path/to/keydir"
+
+# the backend to be used for storing the key, in this case file
+keyring-backend = "file"
+
+# the name of the key you used when importing the key
+key-name = "your-key-name"
+
+# the chain id of the chain to be used
+chain-id = "your-chain-id"
+
+[server-config]
+host = "127.0.0.1"
+port = 9791
+
+[metrics]
+host = "127.0.0.1"
+port = 2113
+```
+
+## 6. Running the Covenant Signer
+
+The covenant signer can be run using the following command:
+
+```shell
+covenant-signer start --config ./path/to/config.toml
+```
+
+the covenant signer must be run in a secure network and only accessible by the 
+covenant emulator.
+
+Once the covenant signer is set up and unlocked, you can configure the covenant 
+emulator to use it. The URL of the covenant signer, which is http://127.0.0.1:9791, 
+should be specified in the covenant emulator's configuration file under the 
+remotesigner section.
+
+It's important to note that the key specified in the covenant emulator's 
+configuration is not the covenant key itself. Instead, it is a 
+key used for sending Cosmos transactions.
+
```

Comment:
We have never specified what this program is and why it exists. We should have an intro that describes the program as:
- a signer program, intended to be used by members of the covenant committee
- intended to contain only the bare minimum as it has access to the private key -- it should be highly secured
- the application logic for the covenant member is implemented covenant-emulator program that interacts with the Babylon blockchain to receive things to sign and forwards them to the covenant signer to receive sigs.